### PR TITLE
K8s 1.12

### PR DESCRIPTION
**Pull request by Arau**
This is to be merged to Master asap as it mentions the 1.12 CSI installation breaking changes.

### CODE REVIEW

**Comment by Arau** [comment]
This is to be merged to Master asap as it mentions the 1.12 CSI installation breaking changes.

File: _docs/platforms/kubernetes/install/1.10.md [hunk 1]
```diff
@@ -18,10 +18,10 @@ cd storageos/k8s/deploy-storageos/standard
 
 or using the Helm chart:
 ```bash
-git clone https://github.com/storageos/helm-chart.git storageos
-cd storageos
+helm repo add storageos https://charts.storageos.com
+helm repo update
 # Set cluster.join to hostnames or ip addresses of at least one node
-helm install . --name my-release --set cluster.join=node01,node02,node03
+helm install storageos/storageos --name my-release --namespace storageos --set cluster.join=node01,node02,node03
```
Comment (darkowlzz) [inline _docs/platforms/kubernetes/install/1.10.md]: I think all these helm commands should start with `$`, as the following commands have `$`.
I know those are notes from the helm chart, but it looks weird to have $ at some places and not at others.

File: _docs/platforms/kubernetes/install/1.10.md [hunk 2]
```diff
@@ -7,7 +7,7 @@ redirect_from: /docs/install/schedulers/kubernetes
 redirect_from: /docs/install/kubernetes
 ---
 
-# Kubernetes 1.10+
+# Kubernetes 1.10 and 1.11
```
Comment (darkowlzz) [inline _docs/platforms/kubernetes/install/1.10.md]: Maybe add a link to the system configurations page after this and a note that linux modules must be installed before following the following instructions.

File: _docs/platforms/kubernetes/install/1.12.md [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+$ ApiAddress=$(echo -n "tcp://$ClusterIP:5705" | base64)
+$ kubectl patch secret/storageos-api --namespace storageos --patch "{\"data\":{\"apiAddress\": \"$ApiAddress\"}}"
+```
+
+## Installation with CSI (Container Storage Interface)
+
+CSI allows you to set StorageOS features (`storageos.com/*` labels) in
+the StorageClass, but not PVCs. If you need to set labels in PVCs (but not
+StorageClasses) or your environment does not support CSI, you may install
+StorageOS without CSI:
```
Comment (darkowlzz) [inline _docs/platforms/kubernetes/install/1.12.md]: I'm not sure if we should mention this here. Maybe in some troubleshooting page, or maybe below the following commands.

**Comment by actionbuddha** [comment]
Slightly confusing wording - we have a heading 'Installation with CSI', and then we proceed to explain out to install without CSI. I'd recommend we start the doc with a paragraph explaining the two choices, which we recommend for general purposes (title 'Overview' or similar), and then give the two variants as installation options.

**Comment by actionbuddha** [comment]
The headings for versions under install are a little inconsistent. At the risk of duplicating content, could we do one page for each of the versions we support? I think it would look a little neater. Also need to ensure that 'v' for version is either present on all, or not at all (probably not at all).

**Comment by Arau** [comment]
> Slightly confusing wording - we have a heading 'Installation with CSI', and then we proceed to explain out to install without CSI. I'd recommend we start the doc with a paragraph explaining the two choices, which we recommend for general purposes (title 'Overview' or similar), and then give the two variants as installation options.

We don't. 'Installation with CSI' is followed by a csi installation

**Comment by Arau** [comment]
> The headings for versions under install are a little inconsistent. At the risk of duplicating content, could we do one page for each of the versions we support? I think it would look a little neater. Also need to ensure that 'v' for version is either present on all, or not at all (probably not at all).

Agreed, I apply these change

File: _docs/platforms/kubernetes/install/1.10.md (hunk 1 above)
Comment (Arau) [reply _docs/platforms/kubernetes/install/1.10.md]: Every command that can be copy and paste from the docs shouldn't have a $ at the beginning to make easier for the user to run the commands.

File: _docs/platforms/kubernetes/install/1.10.md (hunk 2 above)
Comment (Arau) [reply _docs/platforms/kubernetes/install/1.10.md]: I added a note warns people to follow the prerequisites before installing.

File: _docs/platforms/kubernetes/install/1.12.md (hunk 3 above)
Comment (Arau) [reply _docs/platforms/kubernetes/install/1.12.md]: We will add  post GA.

File: _docs/platforms/kubernetes/install/1.10.md (hunk 2 above)
Comment (Arau) [reply _docs/platforms/kubernetes/install/1.10.md]: done

**Comment by Arau** [comment]
Thank you for the changes. They are a significant improvement.

**Pull request merged**